File: core/src/main/java/no/elg/infiniteBootleg/world/ServerClientWorld.java
package no.elg.infiniteBootleg.world;

import no.elg.infiniteBootleg.protobuf.ProtoWorld;
import no.elg.infiniteBootleg.server.ServerClient;
import org.jetbrains.annotations.NotNull;

public class ServerClientWorld extends ClientWorld {

  @NotNull private final ServerClient serverClient;

  public ServerClientWorld(
      @NotNull ProtoWorld.World protoWorld, @NotNull ServerClient serverClient) {
    super(protoWorld);
    this.serverClient = serverClient;
  }

  @NotNull
  public ServerClient getServerClient() {
    return serverClient;
  }
}

File: core/src/main/kotlin/no/elg/infiniteBootleg/world/ServerClientWorld.kt
package no.elg.infiniteBootleg.world

import no.elg.infiniteBootleg.protobuf.ProtoWorld
import no.elg.infiniteBootleg.server.ServerClient

class ServerClientWorld(protoWorld: ProtoWorld.World, val serverClient: ServerClient) : ClientWorld(protoWorld)
